File: CalculatePay.java
import javax.swing.JOptionPane;

public class CalculatePay {
    public static void main( String[] args ) {
        String InputString = JOptionPane.showInputDialog("What is your hourly pay?");
        float HourlyPR = Float.parseFloat(InputString);
        InputString = JOptionPane.showInputDialog("How many hours have you worked?");
        float HoursW = Float.parseFloat(InputString);
        float GrossPay = HourlyPR * HoursW;
        float WitholdongT = 0.0f;
        if (GrossPay > 0 && GrossPay <= 300.00f)
        {
            WitholdongT = 0.10f * GrossPay;
        }
        else if (GrossPay > 300.00f)
        {
            WitholdongT = 0.12f * GrossPay;
        }
        else
        {
            JOptionPane.showMessageDialog(null, "Something went wrong.");
        }
        float NetPay = GrossPay - WitholdongT;
        JOptionPane.showMessageDialog(null, "Your gross pay is: " + GrossPay + "\nYour witholding tax is: " + WitholdongT + "\nYour net pay is: " + NetPay);
    }
}
    

File: C/CalculatePay.c
#include <stdio.h>

int main() {
    puts("What is your hourly pay?");
    float HourlyPR;
    scanf("%f", &HourlyPR);
    puts("How many hours have you worked?");
    float HoursW;
    scanf("%f", &HoursW);
    float GrossPay = HourlyPR * HoursW;
    float WitholdongT = 0.0f;
    if (GrossPay > 0 && GrossPay <= 300.00f)
    {
        WitholdongT = 0.10 * GrossPay;
    }
    else if (GrossPay <= 400.00f)
    {
        WitholdongT = 0.12 * GrossPay;
    }
    else if (GrossPay <= 500.00f)
    {
        WitholdongT = 0.15 * GrossPay;
    }
    else if (GrossPay > 500.00f)
    {
        WitholdongT = 0.20 * GrossPay;
    }
    else
    {
        puts("Something went wrong");
    }
    float NetPay = GrossPay - WitholdongT;
    printf("Your gross pay is: %.2f\n", GrossPay);
    printf("Your witholding tax is: %.2f\n", WitholdongT);
    printf("Your net pay is: %.2f\n", NetPay);
    return 0;
}